opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java

```diff
@@ -267,4 +276,68 @@ public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
                 .build();
         actionList.put(context.getString(R.string.anc_home_visit_observations_n_illnes), observation);
     }
+
+    private void evaluateProblemSolving() throws Exception {
+        problemSolingEvaluationResults.put(context.getString(R.string.child_problem_solving), false);
+        class ChildHVProblemSolvingHelper extends HomeVisitActionHelper {
+            private String child_playing_challenge;
+
+            @Override
+            public void onPayloadReceived(String s) {
+                try {
+                    JSONObject jsonObject = new JSONObject(s);
+                    child_playing_challenge = JsonFormUtils.getValue(jsonObject, "child_playing_challenge");
+                    if (child_playing_challenge.equalsIgnoreCase("hapana") || child_playing_challenge.equalsIgnoreCase("no"))
+                        problemSolingEvaluationResults.put(context.getString(R.string.child_problem_solving), true);
+                    else
+                        problemSolingEvaluationResults.put(context.getString(R.string.child_problem_solving), false);
+                } catch (JSONException e) {
+                    Timber.e(e);
+                }
+            }
+
+            @Override
+            public String evaluateSubTitle() {
+                child_playing_challenge = getYesNoTranslation(child_playing_challenge);
+                return MessageFormat.format("{0}: {1}", context.getString(R.string.child_problem_solving_task), child_playing_challenge);
+            }
+
+            public String getYesNoTranslation(String subtitleText) {
+                if ("yes".equals(subtitleText)) {
+                    return context.getString(R.string.yes);
+                } else if ("no".equals(subtitleText)) {
+                    return context.getString(R.string.no);
+                } else {
+                    return subtitleText;
+                }
+            }
+
+            @Override
+            public String postProcess(String jsonPayload) {
+                return super.postProcess(jsonPayload);
+            }
+
+            @Override
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(child_playing_challenge)) {
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                }
+
+                if (StringUtils.isNotBlank(child_playing_challenge)) {
+                    return BaseAncHomeVisitAction.Status.COMPLETED;
+                } else {
+                    return BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+                }
```

Comment:
You can also show PARTIALLY_COMPLETED status when the parent is experiencing challenges in communicating with her child